sys/net/gnrc/network_layer/ipv6/util.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <assert.h>
+
+#include "log.h"
+#include "net/arp.h"
+#include "net/gnrc/ipv6/nib/pl.h"
+#include "net/gnrc/sixlowpan/ctx.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/rpl.h"
+#include "net/sock.h"
+#include "timex.h"
+
+#include "net/gnrc/netif/internal.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+static bool _ctx_match(const gnrc_sixlowpan_ctx_t *ctx,
+                       const ipv6_addr_t *prefix, uint8_t prefix_len)
+{
+    return (ctx != NULL) &&
+           (ctx->prefix_len == prefix_len) &&
+           (ipv6_addr_match_prefix(&ctx->prefix, prefix) >= prefix_len);
+}
+
+static void _update_6ctx(const ipv6_addr_t *prefix, uint8_t prefix_len,
```

Comment:
Setting the context is typically only something you want to do on the border router, so putting that into the generic code seem misplaced.

Reply:
This is already behind a `if (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_6LBR))`

Reply:
OK, then this should be documented...